rocAL/rocAL/include/meta_data/coco_meta_data_reader.h

```diff
@@ -37,21 +38,32 @@ class COCOMetaDataReader: public MetaDataReader
     void release() override;
     void print_map_contents();
     bool set_timestamp_mode() override { return false; }
-
+    std::pair<uint32_t,uint32_t> get_max_size() override { return std::make_pair(_max_height,_max_width); }
     const std::map<std::string, std::shared_ptr<MetaData>> & get_map_content() override { return _map_content;}
     COCOMetaDataReader();
 private:
     pMetaDataBatch _output;
     std::string _path;
     int meta_data_reader_type;
     void add(std::string image_name, BoundingBoxCords bbox, Labels labels, ImgSize image_size, uint image_id = 0);
-    void add(std::string image_name, BoundingBoxCords bbox, Labels labels, ImgSize image_size, MaskCords mask_cords, std::vector<int> polygon_count, std::vector<std::vector<int>> vertices_count); // To add Mask coordinates to Metadata struct
+    void add(std::string image_name, BoundingBoxCords bbox, Labels labels, ImgSize image_size, MaskCords mask_cords, std::vector<int> polygon_count, std::vector<std::vector<int>> vertices_count, MetaDataType meta_data_type); // To add Mask coordinates to Metadata struct
     bool exists(const std::string &image_name) override;
+    void generate_pixelwise_mask(std::string filename, RLE* R);
     std::map<std::string, std::shared_ptr<MetaData>> _map_content;
     std::map<std::string, std::shared_ptr<MetaData>>::iterator _itr;
     std::map<std::string, ImgSize> _map_img_sizes;
     std::map<std::string, ImgSize> ::iterator itr;
-    std::map<int, int> _label_info;
+    std::map<int, int> _label_info = {{1,1},{2,2},{3,3},{4,4},{5,5},{6,6},{7,7},{8,8}, \
```

Comment:
Hardcoding the contiguous label mapping for coco is not preferred. Use a for loop to generate the mapping in meta_data_reader.cpp where its needed

Reply:
This label generation can't be done on-fly like mask in semantic and it needs this information while storing RLE value